aodndata/aatams_sattag_qc_dm/csv_schema.py

```diff
@@ -0,0 +1,279 @@
+
+        This function can load the csv
+        file by bulk_loading or by reading row by row
+        with skip support.
+
+        Args:
+          fname(str): File name
+          dialect(dict): The csv dialect
+          bulk_load(bool): True for bulkload, False to read row by row
+          skip_every(int): read row interval
+
+        Returns:
+          list: row_numbers - list of row number read
+          dict: header - The header dict
+          list: data_rows - list with data rows
+
+        Raises:
+          csv.Error: if invalid csv file.
+
+        """
+        if dialect is None:
+            dialect = DIALECT_CSV
+
+        with open(fname) as file:
+            reader = csv.reader(file, **dialect)
+            header = next(reader)
+            if bulk_load:
+                data_rows = list(reader)
+                row_numbers = range(1, len(data_rows))
+                if skip_every:
+                    data_rows = data_rows[::skip_every]
+                    row_numbers = row_numbers[::skip_every]
+            else:
+                data_rows = []
+                row_numbers = []
+                for nrow, row in enumerate(reader):
+                    if skip_every:
+                        if nrow % skip_every:
+                            data_rows += [row]
+                            row_numbers += [nrow]
+                    else:
+                        data_rows += [row]
+                        row_numbers += [nrow]
+            return row_numbers, header, data_rows
+
+    @staticmethod
+    def aggregate_to_dict(headers, data_list):
+        """Aggregate all csv rows into a dictionary with header column names as keys and
+        values as a list of all row entries.
+
+        Args:
+          headers(list): a list of strings for keys
+          data_list(list): a list of strings with all data rows
+
+        Returns:
+          Dict[str,List[Any]]: A dict with header columns as keys
+          Dict[str,List[Any]]: A dict with header columns as keys
+          and all respective values in a list.
+
+        """
+        adict = {cname: [] for cname in headers}
+        for data in data_list:
+            for lind, key in enumerate(headers):
+                adict[key] += [data[lind]]
+        return adict
+
+    @staticmethod
+    def compute_cross_values(datadict, keys=None):
+        """Compute a cross value dictionary {k:set(v)}.
+
+        This dictionary is useful to compare
+        if a value of a key is within a group of values.
+
+        Args:
+          datadict(dict): the original dict.
+          keys():
+
+          keys: the selected keys.
+
+        Returns:
+
+        """
+        cross_values = {}
+        if set(keys).issubset(datadict.keys()):
+            cross_values = {x: set() for x in keys}
+            _ = [cross_values[x].update(datadict[x]) for x in cross_values.keys()]
+        return cross_values
+
+    @classmethod
+    def report(cls, msg):
+        """Report a msg.
```

Comment:
Because I hoped that the logger will capture stdout of child classes.
Note this is not a handler class, but a class used within the handler. I will have to look if I need to pass the logger state from the handler to this or if it's just an import.